libevm/VM.cpp

```diff
@@ -233,6 +233,24 @@ void VM::interpretCases()
 		}
 		BREAK
 
+		CASE(REVERT)
+		{
+			// Pre-metropolis
+			if (!m_schedule->haveRevert)
+				throwBadInstruction();
+
+			m_newMemSize = memNeed(*m_SP, *(m_SP - 1));
+			updateMem();
+			ON_OP();
+			updateIOGas();
+
+			uint64_t b = (uint64_t)*m_SP--;
+			uint64_t s = (uint64_t)*m_SP--;
+			owning_bytes_ref output{ move(m_mem), b, s };
```

Comment:
Exception is not ideal, since it's not really an exceptional situation, just another valid opcode.
But I think the code is cleaner this way (and after all REVERT is conceptually described as 'cheap throw')